### PR TITLE
ENH: move to ruff

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@riseup.net>

### CODE REVIEW

**Comment by FFY00** [comment]
This actually caught a bad test!

File: tests/test_wheel.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def test_contents(package_library, wheel_library):
         re.escape('library-1.0.0.dist-info/RECORD'),
         re.escape('library-1.0.0.dist-info/WHEEL'),
         re.escape('library.libs/libexample.so'),
-    ]):
+    ], strict=True):
```
Comment (dnicolodi) [inline tests/test_wheel.py:101]: `zip` has a `strict` argument only starting from Python 3.10.

File: mesonpy/_introspection.py [hunk 2]
```diff
@@ -20,8 +20,8 @@ def debian_python() -> bool:
         import distutils
         try:
             import distutils.command.install
-        except ModuleNotFoundError:
-            raise ModuleNotFoundError('Unable to import distutils, please install python3-distutils')
+        except ModuleNotFoundError as exc:
+            raise ModuleNotFoundError('Unable to import distutils, please install python3-distutils') from exc
```
Comment (dnicolodi) [inline mesonpy/_introspection.py:24]: I don't get the purpose of this change. AFAIK, within the handling of an exception, the `raise ... from` form is exactly the same as the one omitting the `from`. If anything, here I would use `... from None`.

File: mesonpy/_tags.py [hunk 3]
```diff
@@ -60,13 +60,13 @@ def get_abi_tag() -> str:
...
         empty, abi, ext = str(sysconfig.get_config_var('EXT_SUFFIX')).split('.')
-    except ValueError:
+    except ValueError as exc:
         # CPython <= 3.8.7 on Windows does not implement PEP3149 and
         # uses '.pyd' as $EXT_SUFFIX, which does not allow to extract
         # the interpreter ABI.  Check that the fallback is not hit for
         # any other Python implementation.
         if sys.implementation.name != 'cpython':
-            raise NotImplementedError
+            raise NotImplementedError from exc
```
Comment (dnicolodi) [inline mesonpy/_tags.py:69]: Same here.

File: mesonpy/_introspection.py (hunk 2 above)
Comment (FFY00) [reply mesonpy/_introspection.py:24]: From https://github.com/PyCQA/flake8-bugbear#opinionated-warnings

> **B904**: Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling. See [the exception chaining tutorial](https://docs.python.org/3/tutorial/errors.html#exception-chaining) for details.

This is actually very helpful.

File: tests/test_wheel.py (hunk 1 above)
Comment (FFY00) [reply tests/test_wheel.py]: Thanks.

File: mesonpy/_introspection.py (hunk 2 above)
Comment (FFY00) [reply mesonpy/_introspection.py:24]: TLDR: It changed the message from

```
...
During handling of the above exception, another exception occurred:
...
```

to

```
...
The above exception was the direct cause of the following exception:
...
```

Comment (dnicolodi) [reply mesonpy/_introspection.py:24]: I never pick up on the different messages. Thanks for the explanation.

**Comment by FFY00** [comment]
@dnicolodi do you still have any questions, or are you okay with merging? I want to open another PR that conflicts with this, so it'd be good to get this in.

**Comment by dnicolodi** [comment]
I'm not big on linters and I don't know ruff, thus I don't have an opinion. Everything else looks good to me.

**Comment by rgommers** [comment]
Everyone who I've seen try `ruff` seems to like it, so +1 from me.

**Comment by FFY00** [comment]
No one said anything against it, so I am gonna merge it.

If there's anything that you think we might have not considered, feel free to open an issue to discuss it.

**Pull request merged**